cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -287,3 +288,78 @@ fn storage_footprint_to_ledger_footprint(foot: &Footprint) -> Result<LedgerFootp
     })
 }
+
+pub(crate) fn compute_bump_footprint_exp_transaction_data_and_min_fee(
+    footprint: LedgerFootprint,
+    _ledgers_to_expire: u32,
+    snapshot_source: &ledger_storage::LedgerStorage,
+) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
+    let read_bytes =
+        calculate_unmodified_ledger_entry_bytes(footprint.read_only.as_vec(), snapshot_source)?;
+    let soroban_resources = SorobanResources {
+        // TODO: is this correct or should we use the footprint?
+        footprint: LedgerFootprint {
+            read_only: Default::default(),
+            read_write: Default::default(),
+        },
+        instructions: 0,
+        read_bytes,
+        write_bytes: 0,
+        extended_meta_data_size_bytes: 2 * read_bytes,
+    };
+    let transaction_resources = TransactionResources {
+        instructions: 0,
+        read_entries: 0,
+        write_entries: 0,
+        read_bytes: soroban_resources.read_bytes,
+        write_bytes: 0,
+        metadata_size_bytes: soroban_resources.extended_meta_data_size_bytes,
+        transaction_size_bytes: 0,
+    };
+    let fee_configuration = get_fee_configuration(snapshot_source)?;
+    let (min_fee, ref_fee) =
+        compute_transaction_resource_fee(&transaction_resources, &fee_configuration);
+    let transaction_data = SorobanTransactionData {
+        resources: soroban_resources,
+        refundable_fee: ref_fee,
+        ext: ExtensionPoint::V0,
+    };
+    Ok((transaction_data, min_fee))
+}
+
+pub(crate) fn compute_restore_footprint_transaction_data_and_min_fee(
+    footprint: LedgerFootprint,
+    snapshot_source: &ledger_storage::LedgerStorage,
+) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
+    let write_bytes =
+        calculate_unmodified_ledger_entry_bytes(footprint.read_write.as_vec(), snapshot_source)?;
+    let soroban_resources = SorobanResources {
+        // TODO: is this correct or should we use the footprint?
+        footprint: LedgerFootprint {
+            read_only: Default::default(),
+            read_write: Default::default(),
+        },
+        instructions: 0,
+        read_bytes: 0,
+        write_bytes,
+        extended_meta_data_size_bytes: 2 * write_bytes,
+    };
+    let transaction_resources = TransactionResources {
+        instructions: 0,
+        read_entries: 0,
+        write_entries: 0,
+        read_bytes: 0,
+        write_bytes: soroban_resources.write_bytes,
+        metadata_size_bytes: soroban_resources.extended_meta_data_size_bytes,
+        transaction_size_bytes: 0,
+    };
+    let fee_configuration = get_fee_configuration(snapshot_source)?;
+    let (min_fee, ref_fee) =
+        compute_transaction_resource_fee(&transaction_resources, &fee_configuration);
+    let transaction_data = SorobanTransactionData {
+        resources: soroban_resources,
+        refundable_fee: ref_fee,
+        ext: ExtensionPoint::V0,
+    };
+    Ok((transaction_data, min_fee))
+}
```

Comment:
@sisuresh @dmkozh could you review the two functions above to confirm the fee computation is correct?